src/v/storage/log_manager.h

```diff
@@ -212,6 +212,11 @@ class log_manager {
 
     storage_resources& resources() { return _resources; }
 
+    /*
+     * Return disk usage information for all logs managed on the current core.
+     */
+    ss::future<usage_report> disk_usage();
```

Comment:
@abhijat yeh it's unfortunate, but it looks like it might be hard because the log manager abort source is being passed down, and I don't think abort source can be const in its current form.

wdyt?

```
/home/nwatkins/src/redpanda/src/v/storage/types.h:342:14: note: candidate constructor not viable: 5th argument ('const ss::abort_source') would lose const qualifier
    explicit compaction_config(
```